hedyorg/hedy
# 🪲 Make print interpret variables before definition as literal strings

Fixes #4376
In levels 2 and 3 the print command does not use quotes for literal strings and at the same time it does accept variables. With this change, when a variable is defined after it is used in a print statement, the print command will not treat it as variable but as literal string.

**How to test**
Run Hedy locally and check that in levels 2 and 3 the following snippet does not produce an errors but only a warning for an unused variable.
```
print some thing is funny
thing is ask something funny
```
In level 4 check that the following snippet produces an access-before-definition error.
```
print 'some' thing 'is funny'
thing is ask something funny
```
